### ISSUE TITLE
Verify java library to access OTRS

### ISSUE THREAD

**Issue by gioppoluca** [CloudOpting/cloudopting-manager#195]
We need to verify the required effort to integrate with OTRS (someone maybe forgot about it).
OTRS has a web service system that has also REST interface:
http://otrs.github.io/doc/manual/admin/stable/en/html/genericinterface.html
We need to see if is possible to use the same users from the CO DB perhaps changing the DB structure
http://otrs.github.io/doc/manual/admin/stable/en/html/external-backends.html#customer-data

To start playing I advise to use this docker image:
https://hub.docker.com/r/appcontainers/otrs/
I tested it in the local mode and it works.
There is the chance to just create tickets, is not possible to create queues or users or other so that is the max.
Pretty easy in any case

**Comment by gioppoluca** [comment]
Check if you find in internet some other rest api for adding queues

**Comment by gioppoluca** [comment]
Ping on this

**Comment by guidospadotto-profesia** [comment]
@ClaudioSola is studying all the stuff required to implement this: Docker, OTRS, APIs. It will take a while as they are all huge "beasts".

**Comment by ClaudioSola** [comment]
Hi,
OTRS is working (configured on db and start scheduler) in the docker.

To implement the part of the web interface I'm taking example by:
https://github.com/gtudan/OTRS-Client .

**Comment by gioppoluca** [comment]
Can we have a feedback on what you are doing?
We need to agree on development.
Thanks

**Comment by xeviscc** [comment]
@ClaudioSola Any update on this issue? Did you ended the implementation you mentioned on comment from the 15 of February?

**Comment by ClaudioSola** [comment]
Hi,
No, there are no particular news.
The implementation is not completed.
I have Installed Docker  and configured the image containing OTRS, I have created the first prototype java that allows me to communicate with the service and I started working on the component 'otrs-Component'.
When I have just a moment of time, I end the implementation.

**Comment by ildave** [comment]
I am working on it, I am collecting informations about OTRS API and if they can be used to integrate the ticket management in Cloudopting.

**Comment by ildave** [comment]
After a brief talk with @gioppoluca, we arrived at the conclusion that it is not possible to integrate OTRS with CloudOpting in a meaningful and useful way, because of the lack of OTRS API. OTRS does not expose some key feature that are needed to manage it from another system (for example, creation of Agents, Groups and Ticket Queues).

We decided to investigate in two other directions:

- search for other open source software that could offer a better integration
- define an offline workflow for ticket management, outside CloudOpting, using OTRS

I'm on it, I will report updates.
